Gem/Code/Source/Components/Multiplayer/GemSpawnerComponent.cpp

```diff
@@ -211,60 +235,42 @@ namespace MultiplayerSample
         m_queuedForRemovalGems.erase(entityId);
     }
 
-    AZ::Crc32 GemSpawnerComponentController::ChooseGemType(const LmbrCentral::Tags& tags)
+    AZ::Crc32 GemSpawnerComponentController::ChooseGemType(AZStd::vector<GemSpawnEntry>& gemSpawnList, const LmbrCentral::Tags& tags)
     {
-        if (GetSpawnTablesPerRound().empty())
-        {
-            return {};
-        }
-
-        // Get the current round's spawn table, or the last defined round as a fallback.
-        const uint16_t round = GetNetworkMatchComponentController()->GetRoundNumber();
-        const RoundSpawnTable* table;
-        if (round < GetSpawnTablesPerRound().size())
-        {
-            table = &GetSpawnTablesPerRound()[round];
-        }
-        else
-        {
-            table = &GetSpawnTablesPerRound().back();
-        }
-
         // Calculate the total weight of all applicable gem tags.
         float totalWeight = 0.f;
-        for (const GemWeightChance& gemWeight : table->m_gemWeights)
+        for (GemSpawnEntry& entry : gemSpawnList)
         {
-            const auto tagIterator = tags.find(AZ::Crc32(gemWeight.m_tag));
+            const auto tagIterator = tags.find(entry.m_tag);
+            entry.m_entityHasTag = tagIterator != tags.end();
             if (tagIterator != tags.end())
             {
-                totalWeight += gemWeight.m_weight;
+                totalWeight += entry.m_weight;
             }
         }
         
         // Create a random float in the range of [0, totalWeight)
         float randomWeight = GetNetworkRandomComponentController()->GetRandomFloat() * totalWeight;
 
-        AZ::Crc32 chosenType(table->m_gemWeights.front().m_tag);
-        for (const GemWeightChance& gemWeight : table->m_gemWeights)
+        AZ::Crc32 chosenType;
+        for (GemSpawnEntry& entry : gemSpawnList)
         {
-            const auto tagIterator = tags.find(AZ::Crc32(gemWeight.m_tag));
-            if (tagIterator == tags.end())
+            if (entry.m_entityHasTag)
             {
-                continue;
-            }
+                // For every acceptable tag, reduce the the weight value until the right gem type is found for the random value.
```

Comment:
diagrams :D

Reply:
Yah, +1 to whomever put this one there in the first place. :)